docs/tutorials/quickstart.rst

```diff
@@ -0,0 +1,109 @@
+* `VirtualBox <https://www.virtualbox.org/>`_: a software virtualization tool
+* `Vagrant <https://www.vagrantup.com/downloads.html>`_: a command line utility for managing the lifecycle of virtual machines
+
+As the focus of this tutorial is NAPALM, we don't even scratch the surface of these tools.  If you're not familiar with them, please do some research [#f1]_ as they will be an important part of your development/ops toolkit.
+
+Installation
+------------
+
+Install NAPALM with pip::
+
+    pip install napalm
+
+Arista vEOS
+-----------
+
+We'll use Arista, which can be downloaded for free from the Arista site.
+
+Create an account at https://www.arista.com/en/user-registration, and go to https://www.arista.com/en/support/software-download.
+
+Download the latest "vEOS-lab-<version>-virtualbox.box" listed in the vEOS folder at the bottom of the page.
+
+Add it to your vagrant box list, changing the `<version>`::
+
+    $ vagrant box add --name vEOS-lab-<version>-virtualbox ~/Downloads/vEOS-lab-<version>-virtualbox.box
+    $ vagrant box list
+    vEOS-lab-quickstart (virtualbox, 0)
+
+You can delete the downloaded .box file once you have added it, as ``vagrant box add`` copies downloaded file to a designated directory (e.g., for Mac OS X and Linux: ``~/.vagrant.d/boxes``, Windows: ``C:/Users/USERNAME/.vagrant.d/boxes``).
+
+Starting Vagrant
+----------------
+
+Create a Vagrantfile on your machine with the following content:
+
+.. literalinclude:: Vagrantfile
+   :language: ruby
+
+The above content is also available on `GitHub <https://raw.githubusercontent.com/napalm-automation/napalm/master/docs/tutorials/Vagrantfile>`_.
+
+This Vagrantfile creates a base box and a vEOS box when you call "vagrant up"::
+
+    $ vagrant up
+    ... [output omitted] ...
+
+    $ vagrant status
+    Current machine states:
+    base                      running (virtualbox)
+    eos                       running (virtualbox)
+
+You may see some errors when the eos box is getting created [#f2]_.
+
+
+Using the NAPALM command-line client
+------------------------------------
+
+You can now try the example commands at http://napalm.readthedocs.org/en/latest/cli.html, using the test data files at https://github.com/napalm-automation/napalm-eos/tree/master/test/unit/eos::
+
+    # Get a sample test file (or copy/paste it from GitHub to new_good.conf):
+    $ wget https://raw.githubusercontent.com/napalm-automation/napalm-eos/master/test/unit/eos/new_good.conf
+
+    # dry run.
+    # (When prompted, the password is "vagrant")
+    $ cl_napalm_configure \
+      --user vagrant \
+      --vendor eos \
+      --strategy replace \
+      --optional_args 'port=12443' \
+      --dry-run \
+      new_good.conf localhost
+
+
+Try the other examples and commands given at http://napalm.readthedocs.org/en/latest/cli.html.
+
+Shutting down
+-------------
+
+    $ cd PROJECT_ROOT/docs/tutorials
```

Comment:
Right, I needed a way for people to get back to the folder with the Vagrantfile, typing "vagrant destroy" in any other directory doesn't work.  This was the cleanest way I could think of to write it.

I would recommend keeping the tutorial in alongside the rest of the code, rather than as a separate repo.  Ideally, the tutorial instructions should be tested alongside the code, to ensure that the instructions stay current.  By "testing" here, I mean actual automated testing, and not relying on people to remember to test things when they change code.  This isn't actually done here yet, we could backlog that idea.

@dbarrosop - can we merge this into master?  I have two reasons for the request:

1) It looks like we're talking about improvements that we can make iteratively.  "Release early, release often" is a good thing to follow, sometimes.  Also, once it's in, everyone can make contributions to it and it could go faster.
2) we're talking about changing some of the other tutorials, and I don't want the scope to creep.  Small PRs are good PRs!
3) A colleague was asking for this quickstart material, if this gets into master and onto "readthedocs" it will help our team. :-)

If you would rather keep this separate, I can continue to iterate on this, but I'd prefer to merge it and add issues that I'll deal with quickly in separate smaller PRs.